File: poo/java/Estacionamento/Mensalista.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 11/04/14
 *  R.A.: 86492
 */

public class Mensalista extends Veiculo{

    /* Atributos */
    private double valorMensal;

    /* Construtor */
    public Mensalista(String placa, String cor, String modelo,
                      double valorMensal){
        super.setPlaca(placa);
        super.setCor(cor);
        super.setModelo(modelo);
        this.setValorMensal(valorMensal);
    }

    /* Getters */
    public double getValorMensal(){
        return this.valorMensal;
    }

    /* Setters */
    public void setValorMensal(double valorMensal){
        this.valorMensal = valorMensal;
    }

    /* Declarado na super-classe */
    public double PrecoPagar(){
        return this.getValorMensal();
    }

    /* Incrementa o total a pagar em 10% */
    public void AumentaPreco(){
        this.setValorMensal(this.getValorMensal() * 1.10);
    }
}

File: poo/python/Estacionamento/Mensalista.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Veiculo import Veiculo

class Mensalista(Veiculo):

    # Atributos #
    __valorMensal = None

    # Construtor #
    def __init__(self, placa, cor, modelo, valorMensal):
        super(Mensalista, self).setPlaca(placa)
        super(Mensalista, self).setCor(cor)
        super(Mensalista, self).setModelo(modelo)
        self.setValorMensal(valorMensal)

    # Getters #
    def getValorMensal(self):
        return self.__valorMensal
    
    # Setters #
    def setValorMensal(self, valorMensal):
        self.__valorMensal = valorMensal

    # Redefinido da super classe #
    def PrecoPagar(self):
        return self.getValorMensal()

    # Incrementa o total a pagar em 10% #
    def AumentaPreco(self):
        self.setValorMensal(self.getValorMensal() * 1.10)
